lib/Sema/TypeCheckType.cpp

```diff
@@ -3453,6 +3472,13 @@ void TypeChecker::checkUnsupportedProtocolType(Decl *decl) {
       decl->getDeclContext()->isModuleScopeContext())
     return;
 
+  // Global variable declarations are okay to skip, because at the top level of script/repl
```

Comment:
So it sounds like you're actually skipping decls nested inside statements or other decls here, because those decls will be visited later?